src/OrleansRuntime/Placement/PlacementDirectorsManager.cs

```diff
@@ -106,19 +106,27 @@ private IActivationSelector ResolveSelector(PlacementStrategy strategy, bool add
 
         private void ResolveBuiltInStrategies()
         {
-            var strategies = new PlacementStrategy[]
+            var statelessWorker = new StatelessWorkerPlacement();
+
+            var placementStrategies = new PlacementStrategy[]
             {
                 RandomPlacement.Singleton,
                 ActivationCountBasedPlacement.Singleton,
-                new StatelessWorkerPlacement(),
+                statelessWorker,
                 PreferLocalPlacement.Singleton
             };
-            foreach (var strategy in strategies)
+
+            foreach (var strategy in placementStrategies)
+                this.ResolveDirector(strategy);
+            
+            var selectorStrategies = new PlacementStrategy[]
             {
-                var director = this.ResolveDirector(strategy);
-	            if (director is IActivationSelector)
-		            this.ResolveSelector(strategy);
-            }
```

Comment:
The problem was rather subtle. `RandomPlacementDirector` implements both `IPlacementDirector` and `IActivationSelector`. `ActivationCountPlacementDirector` extends `RandomPlacementDirector`, but it only really implements `IPlacementDirector`, though the check `is IActivationSelector` returns `true` for it. That was causing a problem.

I considered splitting `RandomPlacementDirector` into separate classes for placement director and activation selector, but thought it would cause even more code churn. Can be done as a separate change later.